### PR TITLE
Mazemap search functionality

### PR DESCRIPTION
**Pull request by hermabe**
Added a widget that implements Mazemap search. Specifying `url_field` when using this widget enables autofill of link to mazemap when a search result is selected.

Made a separate class for the `timeplace` form and use the Mazemap search widget for `place` field.

### CODE REVIEW

File: web/widgets.py [hunk 1]
```diff
@@ -24,3 +24,41 @@ def __init__(self, *args, **kwargs):
...
+        default_attrs["class"] = default_attrs.get("class", "") + " " + required_class_attr
+
+        super().__init__(attrs=default_attrs)
+
+    def get_context(self, name, value, attrs):
+        context = super().get_context(name, value, attrs)
+        context["widget"].update({
+            "campus_id": self.campus_id,
+            "max_results": self.max_results,
+            "url_field": self.url_field,
```
[the remark this answers is not in the document]
Comment (hermabe) [reply web/widgets.py]: If the extra variables are put into `attrs` they will be rendered as html-attributes in the `<input>` element. Can of course put them in `attrs` and reference `widget.attrs.campus_id` instead and ignore the extra html attributes that have no effect. As they are not html attributes I but them in separate fields.

File: web/templates/web/forms/widgets/mazemap_search.html [hunk 2]
```diff
@@ -0,0 +1,53 @@
+{% load i18n %}
+
+<div class="ui search">
+    {% include "django/forms/widgets/text.html" %}
+    <div class="results" style="width: 100%;"></div>
```
[the remark this answers is not in the document]
Comment (hermabe) [reply web/templates/web/forms/widgets/mazemap_search.html]: Yes, that is better.

File: web/templates/web/forms/widgets/mazemap_search.html [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                '&rows={{ widget.max_results }}' +
+                '&campusid={{ widget.campus_id }}',
+
+            // Convert mazemap search results to fit semantic ui search format
+            onResponse: (mazemap_response => ({
+                results: mazemap_response.result.map(item => ({
+                    title: (
+                        item.dispPoiNames[0] +
+                        (item.dispBldNames[0] ? ", " + item.dispBldNames[0] : "")
+                    ).replace(/<[/]?em>/g, ""),
```
[the remark this answers is not in the document]
Comment (hermabe) [reply web/templates/web/forms/widgets/mazemap_search.html]: Can switch to the general one. I don't think there are any names with `<` or `>` in them.

File: news/forms.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+
+class TimePlaceForm(ModelForm):
+    class Meta:
+        model = TimePlace
+        fields = '__all__'
+        exclude = []
+        widgets = {
+            "place": MazemapSearchInput(url_field="place_url")
+        }
```
[the remark this answers is not in the document]
Comment (hermabe) [reply news/forms.py]: Can do.

File: web/widgets.py [hunk 5]
```diff
@@ -24,3 +25,36 @@ def __init__(self, *args, **kwargs):
...
+        :param attrs: HTML attributes for the <input> element
+        """
+
+        default_attrs = {
+            "placeholder": _("Search places"),
+            "data-campusId": campus_id,
+            "data-maxResults": max_results,
+            "data-urlField": url_field,
+        }
+        required_class_attr = "prompt"
```
Comment (zootos) [inline web/widgets.py]: This looks like a class attribute in regards to Python. Would probably be better code quality to define it in this way also. That is moving it to right beneath the `template_name` class attribute.

File: web/static/web/js/mazemap-search.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
+jQuery.fn.extend({
+    mazemapSearch: function () {
+        let data = this.data();
+        this.parent().search({
```
Comment (zootos) [inline web/static/web/js/mazemap-search.js]: Place file in `web/static/web/js/widgets/`?

**Pull request merged**